mos-platform/geos-cbm/vlir.ld

```diff
@@ -0,0 +1,218 @@
+__overlaysize   = DEFINED(__overlaysize_param) ? __overlaysize_param : 0x1000;
+
+__backbufsize	= 0x2000;
+__overlayaddr 	= 0x8000 - __backbufsize - __overlaysize;
+__stacksize   	= 0x0400;
+__stack   	= __overlayaddr;
+__zp_reg_start  = 0x90;
+__zp_reg_end    = 0xfb;
+__zp_reg_size   = __zp_reg_end - __zp_reg_start;
+
+MEMORY {
+  zp          : ORIGIN = __zp_reg_start, LENGTH = __zp_reg_size
+  vlir00(rwx) : ORIGIN = 0x0400, LENGTH = __overlayaddr - __stacksize - 0x0400 
+  vlir01(rwx) : ORIGIN = __overlayaddr+0*__overlaysize, LENGTH = __overlaysize
+  vlir02(rwx) : ORIGIN = __overlayaddr+1*__overlaysize, LENGTH = __overlaysize
+  vlir03(rwx) : ORIGIN = __overlayaddr+2*__overlaysize, LENGTH = __overlaysize
+  vlir04(rwx) : ORIGIN = __overlayaddr+3*__overlaysize, LENGTH = __overlaysize
+  vlir05(rwx) : ORIGIN = __overlayaddr+4*__overlaysize, LENGTH = __overlaysize
+  vlir06(rwx) : ORIGIN = __overlayaddr+5*__overlaysize, LENGTH = __overlaysize
+  vlir07(rwx) : ORIGIN = __overlayaddr+6*__overlaysize, LENGTH = __overlaysize
+  vlir08(rwx) : ORIGIN = __overlayaddr+7*__overlaysize, LENGTH = __overlaysize
+  vlir09(rwx) : ORIGIN = __overlayaddr+8*__overlaysize, LENGTH = __overlaysize
+  vlir0a(rwx) : ORIGIN = __overlayaddr+9*__overlaysize, LENGTH = __overlaysize
+  vlir0b(rwx) : ORIGIN = __overlayaddr+10*__overlaysize, LENGTH = __overlaysize
+  vlir0c(rwx) : ORIGIN = __overlayaddr+11*__overlaysize, LENGTH = __overlaysize
+  vlir0d(rwx) : ORIGIN = __overlayaddr+12*__overlaysize, LENGTH = __overlaysize
+  vlir0e(rwx) : ORIGIN = __overlayaddr+13*__overlaysize, LENGTH = __overlaysize
+  vlir0f(rwx) : ORIGIN = __overlayaddr+14*__overlaysize, LENGTH = __overlaysize
+
+  dir_block(rw)       : ORIGIN = 0x0000, LENGTH = 0xfe
+  info_block(rw)      : ORIGIN = 0x00fe, LENGTH = 0xfe
+  vlir_block_rest(rw) : ORIGIN = 0x01fc, LENGTH = 0xfe - 16 * 2
+}
+
+REGION_ALIAS("c_readonly", vlir00)
+REGION_ALIAS("c_writeable", vlir00)
+
+SECTIONS { 
+  
+  INCLUDE c.ld 
+
+  .dir_block : { *(.dir_block .dir_block.*) } >dir_block
+  .info_block : { *(.info_block .info_block.*) } >info_block
+
+  .vlir01 __overlayaddr : AT (__overlayaddr+0*__overlaysize) { *(.vlir01 .vlir01.*) NOCROSSREFS(.vlir01 .vlir02 .vlir03 .vlir04 .vlir05 .vlir06 .vlir07 .vlir08 .vlir09 .vlir0a .vlir0b .vlir0c .vlir0d .vlir0e .vlir0f) } > vlir01
```

Comment:
It should be possible to break out NOCROSSREFS and only specify it once; I don't think it needs to appear in each output section to work.

See https://sourceware.org/binutils/docs/ld/Miscellaneous-Commands.html#index-NOCROSSREFS_0028sections_0029

Reply:
Found no way. Always get error messages if i put it somewhere else.

Can you give me an example where exactly to put the NOCROSSREFS clause?

Thanks a lot!

Reply:
Ah, I see what's happening. `NOCROSSREFS` actually only works at the top level; it needs to be a sibling of `SECTIONS`. You've defined an input section description for the file `NOCROSSREFS` that matches the input sections `.vlir01`, `.vlir02`, etc. This is a place where lld's linker script parser is very weak; it doesn't really prevent keywords from being used out of context.